### ISSUE TITLE
.attr("name") on DocumentType returns empty string since 1.18.2

### ISSUE THREAD

**Issue by borkdude** [jhy/jsoup#2241]
In version 1.18.2 the `.documentType` on a document as parsed from `"<!DOCTYPE html>"` returns an empty string for the `name` attribute:

``` java
DocumentType dt = Jsoup.parse("<!DOCTYPE html>");
dt.attr("name") // "html" in 1.18.1 but "" in 1.18.2 and .3
```

Is this intentional or a bug? This surfaced when I tried to upgrade Jsoup in the hickory library for Clojure:

https://github.com/clj-commons/hickory

**Comment by borkdude** [comment]
It seems the breaking change was made here:

https://github.com/jhy/jsoup/commit/4690661055276b9ed269bb56e1147070434aed5d#diff-a23756c0ecd21ca15e014227dbbf76ece534fe3afa1139b103a5a5b565ee8137L16

**Comment by jhy** [comment]
Hi Michiel,

I reverted it so that using `.attr("name")` will work again. I had considered these internal attributes and so that the specific key used would not matter (as the defined API to get the name is `doctype.name()`), but effectively, yes, they are part of the public API.

Thanks for raising!

**Comment by borkdude** [comment]
@jhy Sweet! I'll wait for the next release. Thanks so much.

**Comment by borkdude** [comment]
@jhy I tried the new release but this doesn't seem to be fixed in 1.19.1. I'm actually using this library from Clojure but this is the same code as above in Java basically, in a REPL session. I'm still seeing an empty string for `(.attr dt "name")` 

```
user=> (import 'org.jsoup.Jsoup)
org.jsoup.Jsoup
user=> (Jsoup/parse "<!DOCTYPE html>")
#object[org.jsoup.nodes.Document 0x4ba89729 "<!doctype html>\n<html>\n <head></head>\n <body></body>\n</html>"]
user=> (def dt (Jsoup/parse "<!DOCTYPE html>"))
#'user/dt
user=> (.attr dt "name")
""
```

**Comment by borkdude** [comment]
Sorry, I should have written `(.attr (.documentType dt) "name")`. This makes a difference. I'm still seeing some failing tests in hickory and I'll get back to what is breaking.
